evalml/automl/engine/sequential_engine.py

```diff
@@ -1,105 +1,57 @@
-
-logger = get_logger(__file__)
 
 
-class SequentialEngine(EngineBase):
-    """The default engine for the AutoML search. Trains and scores pipelines locally, one after another."""
-
-    def evaluate_batch(self, pipelines):
-        """Evaluate a batch of pipelines using the current dataset and AutoML state.
+class SequentialComputation(EngineComputation):
+    """A Future-like api for jobs created by the SequentialEngine."""
 
-        Arguments:
-            pipelines (list(PipelineBase)): A batch of pipelines to be fitted and evaluated.
+    def __init__(self, work, **kwargs):
+        self.work = work
+        self.kwargs = kwargs
+        self.meta_data = {}
 
-        Returns:
-            list (int): a list of the new pipeline IDs which were created by the AutoML search.
-        """
-        if self.X_train is None or self.y_train is None:
-            raise ValueError("Dataset has not been loaded into the engine.")
-        new_pipeline_ids = []
-        index = 0
-        while self._should_continue_callback() and index < len(pipelines):
-            pipeline = pipelines[index]
-            self._pre_evaluation_callback(pipeline)
-            X, y = self.X_train, self.y_train
-            if pipeline.model_family == ModelFamily.ENSEMBLE:
-                X, y = self.X_train.iloc[self.ensembling_indices], self.y_train.iloc[self.ensembling_indices]
-            elif self.ensembling_indices is not None:
-                training_indices = [i for i in range(len(self.X_train)) if i not in self.ensembling_indices]
-                X = self.X_train.iloc[training_indices]
-                y = self.y_train.iloc[training_indices]
-            evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, X, y)
-            new_pipeline_ids.append(self._post_evaluation_callback(pipeline, evaluation_result))
-            index += 1
-        return new_pipeline_ids
+    def done(self) -> bool:
+        """Is the computation done?"""
+        return True
 
-    def train_batch(self, pipelines):
-        """Train a batch of pipelines using the current dataset.
+    def get_result(self):
+        """Get the computation result.
+        Will block until the computation is finished.
 
-        Arguments:
-            pipelines (list(PipelineBase)): A batch of pipelines to fit.
-        Returns:
-            dict[str, PipelineBase]: Dict of fitted pipelines keyed by pipeline name.
+        Raises Exception: If computation fails. Returns traceback.
         """
-        super().train_batch(pipelines)
+        return self.work(**self.kwargs)
 
-        fitted_pipelines = {}
-        for pipeline in pipelines:
-            try:
-                fitted_pipeline = EngineBase.train_pipeline(
-                    pipeline, self.X_train, self.y_train,
-                    self.automl.optimize_thresholds,
-                    self.automl.objective
-                )
+    def cancel(self) -> None:
+        """Cancel the current computation."""
```

Comment:
Was this purposely left empty?

Reply:
Yea we don't need to cancel the sequential jobs in the same way we would for a dask/other parallel library job so I'm just leaving it empty because codecov doesn't like `pass` statements.